### PR TITLE
[ML] Add subscription scope for live tests

### PR DESCRIPTION
**Pull request by needuv**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/tests.yml [hunk 1]
```diff
@@ -6,10 +6,9 @@ stages:
       ServiceDirectory: ml
       MatrixReplace:
         - TestSamples=.*/true
+      CloudConfig:
+        Public:
+          SubscriptionConfiguration: $(python-ml-subscription-scope-configuration)
       EnvVars:
-        AZURE_CLIENT_ID: $(aad-azure-sdk-test-client-id)
```
Comment (benbp) [inline sdk/ml/tests.yml:10]: FYI you may need `AZURE_CLIENT_ID: $(ML_CLIENT_ID)` and equivalents, unless your tests handle these namespaced env variable lookups already.

**Comment by benbp** [comment]
Also we will need to set up a variable group and test secrets keyvault, and link them to the pipeline. I can help with this. Is the sub config stored somewhere currently or new? CC @scbedd

File: sdk/ml/tests.yml (hunk 1 above)
Comment (needuv) [reply sdk/ml/tests.yml:10]: our tests do look up the environment variables by namespace (i.e. they will look for `ML_...`). I don't need to include these, right?

**Comment by scbedd** [comment]
@benbp Neehar added the sub config to our KV  then I updated the artifact list for resource provisioner variable group to include their sub config. I found our other partner teams sub configs in there as well so I figured it would be the right place to put it.

**Review by scbedd — APPROVED** [review]
I think this is g2g, short of the analyze failure that you can ignore.

File: sdk/ml/tests.yml (hunk 1 above)
Comment (benbp) [reply sdk/ml/tests.yml:10]: Correct, you don't need to include them.

**Pull request merged**